tests/splines/extrapolation_rule.cpp

```diff
@@ -0,0 +1,398 @@
+        (ddc::init_discrete_space(IDimX::
+                                          init(x0<typename IDimX::continuous_dimension_type>(),
+                                               xN<typename IDimX::continuous_dimension_type>(),
+                                               DVect<IDimX>(ncells))),
+         ...);
+        ddc::init_discrete_space<BSplines<typename IDimI1::continuous_dimension_type>>(
+                x0<typename IDimI1::continuous_dimension_type>(),
+                xN<typename IDimI1::continuous_dimension_type>(),
+                ncells);
+        ddc::init_discrete_space<BSplines<typename IDimI2::continuous_dimension_type>>(
+                x0<typename IDimI2::continuous_dimension_type>(),
+                xN<typename IDimI2::continuous_dimension_type>(),
+                ncells);
+#elif defined(BSPLINES_TYPE_NON_UNIFORM)
+        (ddc::init_discrete_space<IDimX>(breaks<typename IDimX::continuous_dimension_type>(ncells)),
+         ...);
+        ddc::init_discrete_space<BSplines<typename IDimI1::continuous_dimension_type>>(
+                breaks<typename IDimI1::continuous_dimension_type>(ncells));
+        ddc::init_discrete_space<BSplines<typename IDimI2::continuous_dimension_type>>(
+                breaks<typename IDimI2::continuous_dimension_type>(ncells));
+#endif
+        ddc::init_discrete_space<IDimI1>(
+                GrevillePoints<
+                        BSplines<typename IDimI1::continuous_dimension_type>>::get_sampling());
+        ddc::init_discrete_space<IDimI2>(
+                GrevillePoints<
+                        BSplines<typename IDimI2::continuous_dimension_type>>::get_sampling());
+    }
+};
+
+// Checks that when evaluating the spline at interpolation points one
+// recovers values that were used to build the spline
+template <typename ExecSpace, typename MemorySpace, typename I1, typename I2, typename... X>
+static void ExtrapolationRuleSplineTest()
+{
+    // Instantiate execution spaces and initialize spaces
+    Kokkos::DefaultHostExecutionSpace host_exec_space = Kokkos::DefaultHostExecutionSpace();
+    ExecSpace exec_space = ExecSpace();
+    std::size_t constexpr ncells = 10;
+    DimsInitializer<
+            IDim<I1, I1, I2>,
+            IDim<I2, I1, I2>,
+            BatchDims<IDim<I1, I1, I2>, IDim<I2, I1, I2>, IDim<X, I1, I2>...>>
+            dims_initializer;
+    dims_initializer(ncells);
+
+    // Create the values domain (mesh)
+    auto interpolation_domain = ddc::DiscreteDomain<IDim<I1, I1, I2>, IDim<I2, I1, I2>>(
+            GrevillePoints<BSplines<I1>>::get_domain(),
+            GrevillePoints<BSplines<I2>>::get_domain());
+    ddc::DiscreteDomain<IDim<X, void, void>...> const dom_vals_tmp = ddc::DiscreteDomain<
+            IDim<X, void, void>...>(
+            ddc::DiscreteDomain<IDim<
+                    X,
+                    void,
+                    void>>(Index<IDim<X, void, void>>(0), DVect<IDim<X, void, void>>(ncells))...);
```

Comment:
```suggestion
    ddc::DiscreteDomain<IDim<X, void, void>...> const dom_vals_tmp(
            ddc::DiscreteDomain<IDim<
                    X,
                    void,
                    void>>(Index<IDim<X, void, void>>(0), DVect<IDim<X, void, void>>(ncells))...);
```

Reply:
`error: expected ')' before '...' token`

Reply:
Works for me

Reply:
I commit to see what CI gives

Reply:
![image](https://github.com/CExA-project/ddc/assets/22033714/18f80f30-adca-4480-b786-efc68f5979c6)

compiles with gcc but not nvcc, i revert it

Reply:
weird ok, can you just put `auto const dom_vals_tmp = ...` instead then